src/Relations/BelongsToMany.php

```diff
@@ -76,7 +76,7 @@ protected function setWhere()
     {
         $foreign = $this->getForeignKey();
 
-        $this->query->where($foreign, '=', $this->parent->getKey());
+        $this->query->where($foreign, '=', $this->parent->{$this->parentKey});
```

Comment:
When a custom `parentKey` is defined, the custom `parentKey`'s value must be searched in the foreign keys list instead of `_id`.